zachsinterview/src/main/java/com/imprev/zachsinterview/NumericalKeyValueFunImpl.java

```diff
@@ -0,0 +1,35 @@
+package com.imprev.zachsinterview;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+public class NumericalKeyValueFunImpl implements NumericalKeyValueFun {
+    
+    private NumericalKeyValueStore keyValueStore;
+    
+    public NumericalKeyValueFunImpl(NumericalKeyValueStore keyValueStore) {
+        this.keyValueStore = keyValueStore;
+    }
+
+    @Override
+    public Integer sumByPrefix(String prefix) {
+        
+        int sum = 0;
```

Comment:
why have sum be an int if the method being called in the impl returns an Integer object.

Reply:
also assigning if the Integer is a null it will cause a runtime exception when assigned to an int